src/python/pants/reporting/reporting.py

```diff
@@ -112,6 +111,18 @@ def initialize(self, run_tracker, all_options, start_time=None):
       raise ValueError(
         "Flags zipkin-trace-id and zipkin-parent-id must both either be set or not set."
       )
+    if trace_id and (len(trace_id) != 16 and len(trace_id) != 32 or \
+      any(ch not in set('0123456789abcdefABCDEF') for ch in trace_id)):
```

Comment:
You can actually use `string.hexdigits` here, rather than writing it down explicitly :)